docs/sql-guide/statements/statement-insert-bulk.md

```diff
@@ -0,0 +1,165 @@
+
+### Map list
+![expr](/assets/images/sql-guide/map_list.svg)
+
+### Transform clause
+![expr](/assets/images/sql-guide/map_list.svg)
+
+### With clause
+![expr](/assets/images/sql-guide/bulk_insert_options.svg)
+![expr](/assets/images/sql-guide/bulk_insert_option.svg)
+
+## DDL syntax
+
+```
+BULK [INSERT | REPLACE]
+  INTO
+    table_name [(column_name,...)]
+  MAP (position type_name,...)
+  [TRANSFORM (expr,...)]
+  FROM
+    ['path/file_name' | 'URL' | 'stream' ]
+  WITH
+    [
+      [BATCHSIZE integer_literal]
+      [ROWSLIMIT integer_literal]
+      [INPUT ['FILE' | 'URL' | 'STREAM']]
+      [FORMAT ['csv' [HEADER_ROW]] | ['ndjson' [ALLOW_MISSING_VALUES]]]
+      ...
+    ]
+```
+
+## Arguments
+
+| Argument | Description | Further information |
+|---|---|---|
+| `INSERT` | Update values in an existing row. |  |
+| `REPLACE` | Replace values in an existing row. |  |
+| `table_name` | Name of target table |  |
+| `column_name` | Valid columns belonging to `table_name`. First column must be defined `_id` column. System builds a column list from existing columns in `table_name` if columns are not specified. |  |
+| `MAP` | MAP defines how the source data is read and the expected data types. Values from the MAP clause are placed directly into the columns specified in the `column_list`. | [Map examples](/docs/sql-guide/statements/statement-insert-bulk/#map-examples)
+| `position` | Ordinal position of value in source. |  |
+| `type_name` | Data type of the value in source. |  |
+| `TRANSFORM expr` | a list of expressions that are evaluated during execution for each row. | [TRANSFORM examples](/docs/sql-guide/statements/statement-insert-bulk/#transform-clause-1) |
+| `FROM` | A single or multi-line string literal that specifies the source of data and are interpreted based on the INPUT option. Quoted string literals must be properly escaped. | [FROM quotation mark examples](/docs/sql-guide/statements/statement-insert-bulk/#from-quotation-examples) |
+| `'path/file_name'` | Valid path and file name for data source. Not available for FeatureBase Cloud. |  |
+| `'URL'` | Valid URL for data source. |  |
+| `'Inline_stream'` | The contents of the literal read as though they were in a file.  | [FROM inline stream examples](/docs/sql-guide/statements/statement-insert-bulk/#from-inline-stream) |
+| `WITH` | Pass one or more statement level options. |  |
+| `BATCHSIZE` | Specify the batch size of the BULK commit. Defaults to 1000. |  |
+| `ROWSLIMIT` | Limit the number of rows processed in a batch. |
+| `INPUT` | Set the type of input to `'FILE'`, `'URL'` or `'STREAM'`. |  |
+| `FORMAT` | Set the format of the source data to `'csv'` or `'ndjson'`. |  |
+| `HEADER_ROW` | Optional CSV argument that will ignore the header in the source CSV file. |  |
+| `ALLOW_MISSING_VALUES` | Optional argument that overrides a `NULL` error message that will stop processing when valid JsonPath expressions in the MAP clause have missing data. |  |
+
+## Errors
```

Comment:
I'm a little hesitant to put this in here since I think it's going to be a huge list. Can we cut this and look to add later potentially so it doesn't hold all of this up?

Reply:
Commented out. I'll recreate at some point as a troubleshooting topic perhaps? Or maybe there's a way to output errors from code with remediation steps?